### PR TITLE
Add hhea

### PR DESCRIPTION
**Pull request by rsheeter**
Seems to fit most naturally alongside hmtx. Matches how loca flies with glyf.

Contributes to #138.

We can now ttx fonts whose hmtx is not entirely made up of long metrics (ref https://github.com/googlefonts/fontmake-rs/pull/156#issuecomment-1471750483):

```shell
$ cargo run -p fontc -- --source resources/testdata/glyphs2/Mono.glyphs
$ ttx -o - build/font.ttf
# marvel at the hhea numberOfHMetrics 1 and the 3 entry hmtx
```

### CODE REVIEW

File: fontc/src/main.rs [hunk 1]
```diff
@@ -801,10 +808,15 @@ impl Workload {
                 }
             }
 
-            // When Glyf merges mark Loca too
+            // Glyf carries Loca along for the ride
             AnyWorkId::Be(BeWorkIdentifier::Glyf) => {
                 self.mark_success(AnyWorkId::Be(BeWorkIdentifier::Loca))
             }
+
+            // Hmtx carries Hmtx along for the ride
```
Comment (anthrotype) [inline fontc/src/main.rs:816]:
```suggestion
            // Hmtx carries hhea along for the ride
```

**Comment by anthrotype** [comment]
FYI https://crates.io/crates/write-fonts/0.1.4 is up

**Review by cmyr — APPROVED** [review]
One QQ inline :)

File: fontbe/src/hmetrics.rs [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+
+        let mut long_metrics: Vec<LongMetric> = static_metadata
+            .glyph_order
+            .iter()
+            .map(|gn| {
+                let glyph = context.get_glyph(gn);
+                let bbox = glyph.bbox();
+                let ir_glyph = context.ir.get_glyph_ir(gn);
+                let Some(ir_instance) = ir_glyph.sources.get(default_location) else {
+                    panic!("{gn} is not defined at the default location.");
```
[the remark this answers is not in the document]
Comment (rsheeter) [reply fontbe/src/hmetrics.rs:44]: Filed to https://github.com/googlefonts/fontmake-rs/issues/164 to make sure we do and to add a friendly way to get it.

**Pull request merged**